geomstats/geometry/product_riemannian_metric.py

```diff
@@ -61,7 +61,10 @@ def inner_product_matrix(self, base_point=None, point_type=None):
         """
         if point_type is None:
             point_type = self.default_point_type
+
+        base_point = gs.to_ndarray(base_point, to_ndim=2)
```

Comment:
why is this necessary? if base_point is only 1d it means there is only one manifold, and this case could be handled in the __init__?

Reply:
nose2 tests.test_product_manifold is failing without it: this is because gs.to_ndarray cannot convert a 1D to a 3D directly, it needs to go through a 2D...

Maybe it would be worth thinking about modifying gs.to_ndarray, however the vectorization PRs might remove a lot of them. Thus, maybe let's postpone this discussion for later?